yawl/lib/yawl/widget/ToggleSwitch.lua

```diff
@@ -28,17 +29,19 @@ function ToggleSwitch:new(parent, x, y, width, height, backgroundColor, foregrou
     checkArg(6, foregroundColor, 'number','nil')
     local o = self.parent(parent, x, y)
     setmetatable(o, {__index = self})
-    o._size = {width = 1, height = 1}
     o._value = false
-    o._visual = Rectangle(o, 1, 1, math.min(width, height), height, foregroundColor or 0)
+    o._slider = {x = 0, width = height, height = height, backgroundColor = foregroundColor}
     o:speed(2)
+    o.backgroundColor = Frame.backgroundColor
```

Comment:
No. Just no.
If you want to add the backgroundColor to the class, don't do it in the constructor